pages/cql/bcs-logic-2018.cql

```diff
@@ -151,11 +226,166 @@ define "Is Mammogram In Last 39 Months":
 	)
 
 define "Is In Administrative Exclusions":
+	"Is Lacking Both Breasts"
+		or ("Is Lacking Left Breast"
+			and "Is Lacking Right Breast")
+
+define "Is Lacking Both Breasts":
 	"Is Bilateral Mastectomy"
+		or "Is History Of Bilateral Mastectomy"
+		or "Is Unilateral Mastectomy With Bilateral Modifier"
+		or "Is Unilateral Mastectomy Twice Spread Two Weeks"
+
+define "Is Lacking Left Breast":
+	"Is Unilateral Mastectomy With Left Modifier"
+		or "Is Unilateral Mastectomy Left"
+		or "Is Absence Of Left Breast"
+
+define "Is Lacking Right Breast":
+	"Is Unilateral Mastectomy With Right Modifier"
+		or "Is Unilateral Mastectomy Right"
+		or "Is Absence Of Right Breast"
 
 define "Is Bilateral Mastectomy":
-	false
-	// TODO: Enumerate the specified alternative criteria for this.
+	exists(
+		[Procedure: "Bilateral Mastectomy Value Set"] Proc
+			where Proc.status.value = 'completed'
+				and end of ChoiceToIntervalOfDT(Proc.performed) before end of "Measurement Period"
+	)
+
+define "Is History Of Bilateral Mastectomy":
+	exists(
+		[Procedure: "History of Bilateral Mastectomy Value Set"] Proc
+			where Proc.status.value = 'completed'
+				and end of ChoiceToIntervalOfDT(Proc.performed) before end of "Measurement Period"
+	)
+	// NOTE: This assumes the history of a procedure is also recorded as a procedure in FHIR
+
+define "Is Unilateral Mastectomy With Bilateral Modifier":
+	exists(
+		[Claim] Claim
+			where Claim.status.value = 'active'
+				and exists(
+				Claim.procedure BbElem
+					with (
+						[Procedure: "Unilateral Mastectomy Value Set"] Proc
+							where Proc.status.value = 'completed'
+								and end of ChoiceToIntervalOfDT(Proc.performed) before end of "Measurement Period"
+					) Proc1
+						such that
+							(if BbElem.procedure is FHIR.Reference
+								then (BbElem.procedure as FHIR.Reference).identifier = Proc1.identifier
```

Comment:
I don't like the identifier equality method being used here. I would just check that they occurred on the same date, which should be sufficient.

where date from BbElem.date = date from start of Proc1

Reply:
The HEDIS spec explicitly said that the Modifier had to be on the same claim for Left/Right hence the identifier equality there; the spec also used identical wording for the Bilateral modifier except for omitting "(same claim)" but I interpreted this that using the same logic for all 3 was reasonable.  Also, same claim doesn't necessarily mean same date.

Reply:
Well, you can't call the equal operator with FHIR.Identifier params... That will result in a runtime error. The engine only knows how to do equivalence on CQL types. You'll need to use Identifier.value or something else.

Reply:
CQL supports a general equality operator for structured types, though it will likely not have the behavior you're expecting. In particular, if any element in either operand is null, the result is null. To get "equality-like" behavior, you can use the equivalent operator (`~`) which does the same comparison but treats nulls as equivalent.